learn/core_concepts/documents.md

```diff
@@ -38,6 +46,30 @@ In the above example, `"id"`, `"title"`, `"genre"`, `"release-year"`, and `"cast
 Each attribute must be associated with a **value**, e.g. `"Kung Fu Panda"` is the value of `"title"`.
 At minimum, the document must contain one field with the **[primary key][primary-key]** attribute and a unique **[document id][document-id]** as its value. Above, that's: `"id": "1564saqw12ss"`.
 
+#### NDJSON objects
+
+NDJSON objects consist of individual lines where each individual line is valid JSON text and each line is delimited with a newline character. Any [rules that apply to formatting NDJSON](http://ndjson.org/) also apply to MeiliSearch documents.
+
+The above JSON document would look like this in NDJSON:
+
+```ndjson
+
+{ "id": "1564saqw12ss", "title": "Kung Fu Panda", "genre": "Children's Animation", "release-year": 2008, "cast": [ {"Jack Black": "Po"}, {"Jackie Chan": "Monkey"} ]}
+```
+
+#### CSV objects
+
+CSV objects are separated by delimiters that could be a space, semicolon, or some other character, though the comma is most common. Any [rules that apply to formatting CSV](https://datatracker.ietf.org/doc/html/rfc4180) also apply to MeiliSearch documents.
+
+The above JSON document would look like this in CSV:
+
+```csv
+  "id","title","genre","release-year","cast"
```

Comment:
Something important is missing about the CSV format. Maybe it should be explained on a dedicated page?

Indeed, the explanation of the typing within the header is missing to help the search engine to select the right type for indexing since the cell values are contained between two quotes.

CSV data format needs to contain a first line representing the list of attributes with the optionally chosen type separated from the attribute name with the `:` character. The type is case insensitive.
An attribute can be specified with two types: string or number. A missing type will be interpreted as a string by default.

Valid headline example: `"id:number","title:string","author","price:number"`

Reply:
@gmourier 
We can have a dedicated page as well, but we do have a link directing the user to the rules for CSV. Is that not enough?

Reply:
Oh, sorry! I didn't see the link. That should be enough! 

We can improve it later if we realize that the information is not as easy to find as we expected. 🤓

I think that it is necessary to have a part that explains the typing of the fields to guide the user who will want to index documents from a CSV file.

Reply:
It could be useful in dedicated guides in the future. e.g. "How to send data from a SQL database to MeiliSearch?". (SQL database exports are often in CSV format).

In addition to providing new formats, the goal is also to speed up indexing times. NDJSON and CSV are streamable. There is no need to load the whole file in RAM to validate it.

At the moment it could be a sentence explaining it succinctly! What do you think?

Reply:
Hmmm…this article is under the `learn` section, so explaining the intricacies of file formats feels a bit too advanced and out-of-scope.

I propose we:

1. leave comments regarding why a user would prefer a file format over another (e.g. CSV over JSON) out of this article;
2. open a new issue proposing a new guide that is specifically about file formats;
3. once the guide is done and published, we'll update `documents.md` with a link to it.